### PR TITLE
{EventHubs} `az eventhubs eventhub`: Add deprecation message to `--message-retention-in-days` 

### PR DESCRIPTION
**Pull request by damodaravadhani**
**Related command**
az eventhubs eventhub create
az eventhubs eventhub update

**Description**
We would be deprecating this parameter in a future release
**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
EventHubs

File: src/azure-cli/azure/cli/command_modules/eventhubs/_params.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def load_arguments_eh(self, _):
 
     for scope in ['eventhubs eventhub update', 'eventhubs eventhub create']:
         with self.argument_context(scope) as c:
-            c.argument('message_retention_in_days', options_list=['--message-retention'], type=int, help='Number of days to retain events for this Event Hub, value should be 1 to 7 days and depends on Namespace sku. if Namespace sku is Basic than value should be one and is Manadatory parameter. Namespace sku is standard value should be 1 to 7 days, default is 7 days and is optional parameter')
+            c.argument('message_retention_in_days', options_list=['--message-retention'], type=int, help='Number of days to retain events for this Event Hub, value should be 1 to 7 days and depends on Namespace sku. if Namespace sku is Basic than value should be one and is Manadatory parameter. Namespace sku is standard value should be 1 to 7 days, default is 7 days and is optional parameter', deprecate_info=c.deprecate(hide=True, expiration='2.46.0'))
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/eventhubs/_params.py:107]: We don't allow breaking changes in `2.46.0`. The next breaking change window would be May 2023. At that time we will release CLI `2.49.0`

File: src/azure-cli/azure/cli/command_modules/eventhubs/custom.py [hunk 2]
```diff
@@ -191,6 +192,8 @@ def cli_keys_renew(client, resource_group_name, namespace_name, name, key_type,
 def cli_eheventhub_create(cmd, client, resource_group_name, namespace_name, event_hub_name, message_retention_in_days=None, partition_count=None, status=None,
                           enabled=None, skip_empty_archives=None, capture_interval_seconds=None, capture_size_limit_bytes=None, destination_name=None, storage_account_resource_id=None, blob_container=None, archive_name_format=None):
     # from azure.mgmt.eventhub.models import Eventhub, CaptureDescription, Destination, EncodingCaptureDescription
+    from argcomplete import warn
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/eventhubs/custom.py:195]: The guideline for log is to import from knack:
```
from knack.log import get_logger
logger = get_logger(__name__)

def foo():
    logger.warning("blablabla")
```

**Comment by damodaravadhani** [comment]
@evelyn-ys , comments are addresed. Can we please merge this PR ? Thanks !

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**